src/components/Menu/Menu/hooks/useCloseMenuOnKeyEvent.tsx

```diff
@@ -1,29 +1,37 @@
-import { useCallback } from "react";
+import React, { useCallback } from "react";
 import useKeyEvent from "../../../../hooks/useKeyEvent";
-import { HIDE_SHOW_EVENTS } from "../../../Dialog/consts/dialog-show-hide-event";
+import { CloseMenuOption } from "../MenuConstants";
+import { keyCodes } from "../../../../constants";
 
 const KEYS = ["Escape", "ArrowLeft"];
 
-export default function useCloseMenuOnKeyEvent(
+export default function useCloseMenuOnKeyEvent({
   hasOpenSubMenu,
   onCloseMenu,
   ref,
   onClose,
   isSubMenu,
   useDocumentEventListeners
-) {
+}: {
+  hasOpenSubMenu: boolean;
+  onCloseMenu: (option: CloseMenuOption) => void;
+  ref: React.RefObject<HTMLElement>;
+  onClose: (option: CloseMenuOption, key?: string) => void;
+  isSubMenu: boolean;
+  useDocumentEventListeners: boolean;
+}) {
   const onEscapeOrLeftArrowClick = useCallback(
-    event => {
+    (event: React.KeyboardEvent) => {
       const isArrowLeftClick = event.key === "ArrowLeft";
 
       if (isArrowLeftClick && !isSubMenu) {
         return;
       }
 
       if (hasOpenSubMenu) return false;
-      onCloseMenu(event);
+      onCloseMenu(event as CloseMenuOption);
```

Comment:
This is super confusing IMO, on the one hand we're calling the argument's type "Options", on the other hand it's a DOM event.

Why not use the built-in type for the event instead?

Reply:
Instead of passing event here, I'm passing now `{propagate: false}` as event doesn't have a `propagate` property